packages/backend-function/src/function_env_translator.ts

```diff
@@ -77,6 +78,17 @@ export class FunctionEnvironmentTranslator {
         return [];
       },
     });
+
+    // Using CDK validation mechanism as a way to generate a typed process.env shim file at the end of synthesis
+    this.lambda.node.addValidation({
+      validate: (): string[] => {
+        new FunctionEnvironmentTypeGenerator(
+          this.lambda.node.id,
+          Object.values(this.ssmEnvVars).map((ssmEnvVar) => ssmEnvVar.name)
```

Comment:
This is only going to include env vars that are derived from SSM. We may need to maintain a new list in this class of all env var names that we append to in the ctor and in `addSsmEnvironmentEntry`

Reply:
Gotcha. Since we are doing the grouping in the typed shim, maybe two new lists: one for defined env vars and another for resource access env vars.

Reply:
For this PR, I'll have a list for defined env vars and the list for resource access env vars will be added when the work is done for that